planner/core/src/main/java/eu/seaclouds/platform/planner/core/Planner.java

```diff
@@ -167,17 +174,19 @@ public String rePlan() throws ParsingException, IOException {
         List<NameValuePair> optParams = getReqParams(
                 new BasicNameValuePair("aam", aam),
                 new BasicNameValuePair("dam", dam),
-                new BasicNameValuePair("mmResults", mmOutput)
+                new BasicNameValuePair("offers", mmOutput)
         );
-        String optResult = optimizerClient.postRequest("/optimize", optParams);
-        return optResult; //TODO: check the output for the optimizer
+
+        Pair<String, String> optResult = optimizerClient.postRequest("/optimize", optParams);
+        if(optResult.first.equals("200")){
+            return optResult.second; //TODO: check the output for the optimizer
+        }
+        return null; //TODO: return failure
     }
 
     public String generateDam() {
 
-        List<NameValuePair> damInfoParams = getReqParams(new BasicNameValuePair("adp", adp));
-
-        String monitorInfo = new HttpHelper(monitorGenURL).postRequest(DAM_GEN_OP, damInfoParams);
+        String monitorInfo = new HttpHelper(monitorGenURL).postInBody(MONITOR_GEN_OP, adp);
 
         if(monitorInfo == null)
             return "error"; //TODO: check it is a good response, need error handling policy
```

Comment:
This will change if SeaClouds adopts a uniform error handling policy. But ok guava is an option.